utils/src/test/java/software/amazon/awssdk/testutils/EnvironmentVariableHelper.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.testutils;
+
+import java.lang.reflect.Field;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+import org.junit.rules.ExternalResource;
+import software.amazon.awssdk.utils.SystemSetting;
+
+/**
+ * A utility that can temporarily forcibly set environment variables and
+ * then allows resetting them to the original values.
+ */
+public class EnvironmentVariableHelper extends ExternalResource {
```

Comment:
Don't we already have this elsewhere?

Reply:
Yea, we already have it in the `test-utils` module, I tried moving it from the `test-utils` into the `utils` module, but there are some other places using this class having the dependency on `test-utils` but not on `utils`, and that caused some issue. So I just copied one here. Should we move it to `utils` and correct all of those usages?

Reply:
Why does this need to be in the utils module?

Reply:
Because we have the `UserHomeDirectoryUtilsTest` using it in the `utils` module. And as we discussed, it's better for us to extract `userHomeDirectory()` method out of `ProfileFileLocation`, thus I moved it into `UserHomeDirectoryUtils`, as well as the `UserHomeDirectoryUtilsTest`.